http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/RuntimeRoutes.scala

```diff
@@ -0,0 +1,142 @@
+import cats.effect.{IO, Timer}
+import cats.mtl.ApplicativeAsk
+import io.circe.Decoder
+import org.broadinstitute.dsde.workbench.leonardo.http.api.LeoRoutes.validateRuntimeNameDirective
+import org.broadinstitute.dsde.workbench.leonardo.http.service.{RuntimeConfigRequest, RuntimeService}
+import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo}
+import org.broadinstitute.dsde.workbench.model.google.{GcsPath, GoogleProject}
+import RuntimeRoutes._
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
+import LeoRoutesJsonCodec.dataprocConfigDecoder
+import JsonCodec._
+import scala.concurrent.duration._
+
+class RuntimeRoutes(runtimeService: RuntimeService[IO], userInfoDirectives: UserInfoDirectives)(
+  implicit timer: Timer[IO]
+) {
+  val routes: server.Route = userInfoDirectives.requireUserInfo { userInfo =>
+    implicit val traceId = ApplicativeAsk.const[IO, TraceId](TraceId(UUID.randomUUID()))
+    // https://doc.akka.io//docs/akka-http/current/routing-dsl/path-matchers.html?language=scala#examples
+    pathPrefix("google" / "v1" / "runtime") {
+      pathPrefix(googleProjectSegment / Segment) { (googleProject, runtimeNameString) =>
+        validateRuntimeNameDirective(runtimeNameString) { runtimeName =>
+          pathEndOrSingleSlash {
+            post {
+              entity(as[CreateRuntime2Request]) { req =>
+                complete(
+                  createRuntimeHandler(
+                    userInfo,
+                    googleProject,
+                    runtimeName,
+                    req
+                  )
+                )
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private[api] def createRuntimeHandler(userInfo: UserInfo,
+                                        googleProject: GoogleProject,
+                                        runtimeName: RuntimeName,
+                                        req: CreateRuntime2Request): IO[ToResponseMarshallable] =
+    for {
+      traceId <- IO(UUID.randomUUID())
+      implicit0(context: ApplicativeAsk[IO, RuntimeServiceContext]) <- timer.clock
+        .realTime(TimeUnit.MILLISECONDS)
+        .map(
+          n =>
+            ApplicativeAsk
+              .const[IO, RuntimeServiceContext](RuntimeServiceContext(TraceId(traceId), Instant.ofEpochMilli(n)))
+        )
+      _ <- runtimeService.createRuntime(
+        userInfo,
+        googleProject,
+        runtimeName,
+        req
+      )
+    } yield StatusCodes.Accepted
+}
+
+object RuntimeRoutes {
+  implicit val gceConfigDecoder: Decoder[RuntimeConfigRequest.GceConfig] = Decoder.forProduct2(
+    "machineType",
+    "diskSize"
+  )((mt, ds) => RuntimeConfigRequest.GceConfig(mt, ds))
+
+  implicit val runtimeConfigDecoder: Decoder[RuntimeConfigRequest] = Decoder.instance { x =>
+    //For newer version of requests, we use `cloudService` field to distinguish whether user is
+    for {
+      cloudService <- x.downField("cloudService").as[CloudService]
```

Comment:
Should `cloudService` be optional and default to GCE?

Reply:
Eh maybe this is fine -- the whole `runtimeConfig` is optional in the request and defaults to GCE… if you specifiy a `runtimeConfig` you should also specify the type.

Reply:
yea, also this is for new routes, so I think it should be okay.